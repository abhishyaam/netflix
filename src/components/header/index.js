import React from 'react';
import { useState } from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';
import {
  Background,
  Container,
  Logo,
  ButtonLink,
  Feature,
  Text,
  FeatureCallOut,
  Link,
  Group,
  Picture,
  Dropdown,
  Profile,
  SearchIcon,
  Search,
  SearchInput,
  PlayButton,
} from './styles/header';

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture src={`/images/users/${src}.png`} {...restProps} />;
};

Header.Search = function HeaderSearch({
  searchTerm,
  setSearchTerm,
  ...restProps
}) {
  const [searchActive, setSeacrhActive] = useState(false);
  return (
    <Search {...restProps}>
      <SearchIcon
        onClick={() => setSeacrhActive((searchActive) => !searchActive)}
      >
        <img src='/images/icons/search.png' alt='Search' />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        placeholder='Search films and series'
        active={searchActive}
      />
    </Search>
  );
};
