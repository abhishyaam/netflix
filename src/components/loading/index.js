import React from 'react';
import { Picture, Spinner, LockBody, ReleaseBody } from './styles/loading';

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner {...restProps}>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} data-testid='loading-picture' />
    </Spinner>
  );
}

Loading.ReleaseBody = function LoadingRealeaseBody({ src, ...restProps }) {
  return <ReleaseBody />;
};
