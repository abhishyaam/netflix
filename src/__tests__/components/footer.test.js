import React from 'react';
import { render } from '@testing-library/react';

import { Footer } from '../../components/';

describe('<Footer />', () => {
  it('renders the <Footer />  with polpulated data', () => {
    const { container, getByText } = render(
      <Footer>
        <Footer.Title>Questions? Contact us</Footer.Title>
        <Footer.Break />
        <Footer.Row>
          <Footer.Column>
            <Footer.Link href='#'>FAQs</Footer.Link>
            <Footer.Link href='#'>Investor Relations</Footer.Link>
            <Footer.Link href='#'>Ways to Watch</Footer.Link>
            <Footer.Link href='#'>Corporate Information</Footer.Link>
            <Footer.Link href='#'>Netflix Originals</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href='#'>Help Centre</Footer.Link>
            <Footer.Link href='#'>Jobs</Footer.Link>
            <Footer.Link href='#'>Terms of Use</Footer.Link>
            <Footer.Link href='#'>Contact Us</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href='#'>Account</Footer.Link>
            <Footer.Link href='#'>Redeem Gift Card</Footer.Link>
            <Footer.Link href='#'>Privacy</Footer.Link>
            <Footer.Link href='#'>Speed Test</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href='#'>Media Centre</Footer.Link>
            <Footer.Link href='#'>Buy Gift Card</Footer.Link>
            <Footer.Link href='#'>Cookie Preferneces</Footer.Link>
            <Footer.Link href='#'>Legal Notices</Footer.Link>
          </Footer.Column>
        </Footer.Row>
        <Footer.Break />
        <Footer.Text>Netflix India</Footer.Text>
      </Footer>
    );

    expect(getByText('Questions? Contact us')).toBeTruthy();
    expect(getByText('FAQs')).toBeTruthy();
    expect(getByText('Investor Relations')).toBeTruthy();
    expect(getByText('Ways to Watch')).toBeTruthy();
    expect(getByText('Corporate Information')).toBeTruthy();
    expect(getByText('Netflix Originals')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
