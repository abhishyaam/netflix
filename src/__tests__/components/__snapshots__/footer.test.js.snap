// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders the <Footer />  with polpulated data 1`] = `
<div
  class="footer__Container-sc-qu6i6-0 bwOijH"
>
  <p
    class="footer__Title-sc-qu6i6-4 euNNgW"
  >
    Questions? Contact us
  </p>
  <div
    class="footer__Break-sc-qu6i6-6 cPBIKh"
  />
  <div
    class="footer__Row-sc-qu6i6-2 fWQFYq"
  >
    <div
      class="footer__Column-sc-qu6i6-1 kXVIYP"
    >
      <a
        class="footer__Link-sc-qu6i6-3 hbjDoA"
        href="#"
      >
        FAQs
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 hbjDoA"
        href="#"
      >
        Investor Relations
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 hbjDoA"
        href="#"
      >
        Ways to Watch
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 hbjDoA"
        href="#"
      >
        Corporate Information
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 hbjDoA"
        href="#"
      >
        Netflix Originals
      </a>
    </div>
    <div
      class="footer__Column-sc-qu6i6-1 kXVIYP"
    >
      <a
        class="footer__Link-sc-qu6i6-3 hbjDoA"
        href="#"
      >
        Help Centre
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 hbjDoA"
        href="#"
      >
        Jobs
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 hbjDoA"
        href="#"
      >
        Terms of Use
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 hbjDoA"
        href="#"
      >
        Contact Us
      </a>
    </div>
    <div
      class="footer__Column-sc-qu6i6-1 kXVIYP"
    >
      <a
        class="footer__Link-sc-qu6i6-3 hbjDoA"
        href="#"
      >
        Account
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 hbjDoA"
        href="#"
      >
        Redeem Gift Card
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 hbjDoA"
        href="#"
      >
        Privacy
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 hbjDoA"
        href="#"
      >
        Speed Test
      </a>
    </div>
    <div
      class="footer__Column-sc-qu6i6-1 kXVIYP"
    >
      <a
        class="footer__Link-sc-qu6i6-3 hbjDoA"
        href="#"
      >
        Media Centre
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 hbjDoA"
        href="#"
      >
        Buy Gift Card
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 hbjDoA"
        href="#"
      >
        Cookie Preferneces
      </a>
      <a
        class="footer__Link-sc-qu6i6-3 hbjDoA"
        href="#"
      >
        Legal Notices
      </a>
    </div>
  </div>
  <div
    class="footer__Break-sc-qu6i6-6 cPBIKh"
  />
  <p
    class="footer__Text-sc-qu6i6-5 izvmYz"
  >
    Netflix India
  </p>
</div>
`;
